flashinfer/sparse.py

```diff
@@ -666,3 +666,522 @@ def run(
     def end_forward(self) -> None:
         r"""Warning: This method is deprecated and has no effect."""
         pass
+
+
+class VariableBlockSparseAttentionWrapper:
+    r"""Wrapper class for attention computation with a block-sparse matrix as attention mask.
+    This API supports variable block sizes provided by ``block_row_sz`` and ``block_col_sz``.
+    Besides, each ``kv_head_idx`` can specify its own sparse patterns without using the same mask.
```

Comment:
I'll appreciate it if you can create documentation page and attach the figure you put in this pull request.

The media files are hosted in https://github.com/flashinfer-ai/web-data/, I have invited you to that repo.